std/groth16_bls12377/verifier.go

```diff
@@ -71,36 +73,65 @@ func Verify(api frontend.API, vk VerifyingKey, proof Proof, publicInputs []front
 		var ki sw_bls12377.G1Affine
 		ki.ScalarMul(api, vk.G1.K[k+1], v)
 		kSum.AddAssign(api, ki)
+
 	}
 
 	// compute e(Σx.[Kvk(t)]1, -[γ]2) * e(Krs,δ) * e(Ar,Bs)
-	ml, _ := sw_bls12377.MillerLoop(api, []sw_bls12377.G1Affine{kSum, proof.Krs, proof.Ar}, []sw_bls12377.G2Affine{vk.G2.GammaNeg, vk.G2.DeltaNeg, proof.Bs})
-	pairing := sw_bls12377.FinalExponentiation(api, ml)
+	pairing, _ := sw_bls12377.Pair(api, []sw_bls12377.G1Affine{kSum, proof.Krs, proof.Ar}, []sw_bls12377.G2Affine{vk.G2.GammaNeg, vk.G2.DeltaNeg, proof.Bs})
 
 	// vk.E must be equal to pairing
 	vk.E.AssertIsEqual(api, pairing)
+
 }
 
 // Assign values to the "in-circuit" VerifyingKey from a "out-of-circuit" VerifyingKey
 func (vk *VerifyingKey) Assign(_ovk groth16.VerifyingKey) {
 	ovk, ok := _ovk.(*groth16_bls12377.VerifyingKey)
 	if !ok {
 		panic("expected *groth16_bls12377.VerifyingKey, got " + reflect.TypeOf(_ovk).String())
+
 	}
 
 	e, err := bls12377.Pair([]bls12377.G1Affine{ovk.G1.Alpha}, []bls12377.G2Affine{ovk.G2.Beta})
 	if err != nil {
 		panic(err)
+
 	}
 	vk.E.Assign(&e)
 
 	vk.G1.K = make([]sw_bls12377.G1Affine, len(ovk.G1.K))
 	for i := 0; i < len(ovk.G1.K); i++ {
 		vk.G1.K[i].Assign(&ovk.G1.K[i])
+
 	}
 	var deltaNeg, gammaNeg bls12377.G2Affine
 	deltaNeg.Neg(&ovk.G2.Delta)
 	gammaNeg.Neg(&ovk.G2.Gamma)
 	vk.G2.DeltaNeg.Assign(&deltaNeg)
 	vk.G2.GammaNeg.Assign(&gammaNeg)
+
+}
+
+// Fill one point for inner verifying key
```

Comment:
maybe a better name would be `Allocate()` with a documentation that explains why this function exists?